js_modules/dagster-ui/packages/ui-core/src/instance/InstanceConcurrency.tsx

```diff
@@ -512,91 +604,111 @@ const DeleteConcurrencyLimitDialog = ({
   );
 };
 
-const ConcurrencyRunsDialog = ({
-  runIds,
+const ConcurrencyActionMenu = ({
+  pendingStep,
+  onUpdate,
+}: {
+  pendingStep: ConcurrencyStepFragment;
+  onUpdate: () => void;
+}) => {
+  const [freeSlots] = useMutation<
+    FreeConcurrencySlotsMutation,
+    FreeConcurrencySlotsMutationVariables
+  >(FREE_CONCURRENCY_SLOTS_MUTATION);
+
+  return (
+    <Popover
+      content={
+        <Menu>
+          <MenuItem
+            key="free-concurrency-slots-step"
+            icon="status"
+            text="Free concurrency slot for step"
+            onClick={async () => {
+              const resp = await freeSlots({
+                variables: {runId: pendingStep.runId, stepKey: pendingStep.stepKey},
+              });
+              if (resp.data?.freeConcurrencySlots) {
+                onUpdate();
+                await showSharedToaster({
+                  intent: 'success',
+                  icon: 'copy_to_clipboard_done',
+                  message: 'Freed concurrency slot',
+                });
+              }
+            }}
+          />
+          <MenuItem
+            key="free-concurrency-slots-run"
+            icon="status"
+            text="Free all concurrency slots for run"
+            onClick={async () => {
+              const resp = await freeSlots({variables: {runId: pendingStep.runId}});
+              if (resp.data?.freeConcurrencySlots) {
+                onUpdate();
+                await showSharedToaster({
+                  intent: 'success',
+                  icon: 'copy_to_clipboard_done',
+                  message: 'Freed concurrency slots',
```

Comment:
Include the number of concurrency slots freed? Also "[X] concurrency slots freed" reads a little nicer to me than having "freed" at the start of the text.